qunicorn_core/core/pilotmanager/ibm_pilot.py

```diff
@@ -344,7 +352,7 @@ def _map_sampler_results_to_dataclass(
                         result_type=ResultType.ERROR.value,
                         job=job,
                         program=programs[i],
-                        result_dict={"exception_message": exception_message},
+                        data={"exception_message": exception_message},
```

Comment:
We should formalize the error type (and maybe move the stacktrace into the data).
Keep in mind that stacktraces (and error messages) can expose information that is better hidden...